CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+Contributions are appreciated and encouraged.
+They can be sumitted via GitHub pull request to
+
+  https://github.com/permon/permon
+
+If you are planning a large contribution, we
+encourage you to discuss the concept by creating an issue and interact
+with us frequently to ensure that your effort is well-directed.
+
+## Guidelines
+PERMON follows PETSc code standards.
+Please read the code standards chapter of the PETSc developer guide https://docs.petsc.org/en/latest/developers/style.html.
+
+Before filing a pull request (PR):
+
+- If your contribution can be logically decomposed into 2 or more separate contributions, submit them in sequence with different branches instead of all at once.
+- Include tests which cover any changes to the source code.
+- Before submitting any PR, run the full test suite - i.e `make alltests TIMEOUT=600` on your machine
+
+### Guidelines for maintainers
+
+- All merges to master shall be with no fast forward - i.e. `--no-ff`
```

Comment:
OK, if you remove this, I'm otherwise fine with this pull request.
```suggestion
```

Reply:
It's actually  a chance for you to at least try the workflow of just clicking `Commit suggestion` above and use squash&merge ;-)